proposals/low-level-struct-improvements.md

```diff
@@ -824,19 +817,69 @@ ref struct S<T> {
 
 ## Open Issues
 
+### Keywords vs. attributes
+This design calls for using attributes to annotate the new lifetime rules. This also could've been done just as easily with contextual keywords. For instance `[DoesNotEscape]` could map to `scoped`. However keywords, even the contextual ones, generally must meet a very high bar for inclusion. They take up valuable langauge real estate and are more prominent parts of the langauge. This feature, while valuable, is going to serve a minority of C# developers.
+
+On the surface that would seem to favor not using keywords but there are two important points to consider: 
+
+1. The annotations will effect program semantics. Having attributes impact program semantics is a line C# is reluctant to cross and it's unclear if this is the feature that should justify the language taking that step.
+1. The developers most likely to use this feature intesect strongly with the set of developers that use function pointers. That feature, while also used by a minority of developers, did warrant a new syntax and that decision is still seen as sound. 
+
+Taken together this means syntax should be considered.
+
+A rough sketch of the syntax would be: 
+
+- `[DoesNotEscape]` maps to `scoped` 
+- `[RefThisEscapes]` maps to `escapes`
+- `[RefFieldsEscapes]` maps to `escapes` but attached to the `ref` of the method.
```

Comment:
We should consider attaching the attribute or keyword to the parameter rather than the method, to allow distinguishing between multiple `ref` parameters.

Reply:
Agree. I was hesistant before because I thought it was overkill. Subsequent discussions though lead me to think otherwise. It's one of the changes I'm going to make after LDM today. I didn't want to rewrite the doc though until I had a sense of the syntax we will use (don't want to rewrite twice essentially). This section is rewritten with the assumption it will be on individual parameters